lesson18/main.js

```diff
@@ -150,6 +161,43 @@ const getCurrentPageNum = () => {
   pagination.textContent = `${getCurrentImgIndex() + 1}/${imgArray.length}`;
 };
 
+const createDotPagination = () => {
+  const pagination = createElementWithClassName("div", "dot-pagination");
+  const fragment = document.createDocumentFragment();
+  for (let i = 0; i < imgArray.length; i++) {
+    const dot = createElementWithClassName("sapn", "dot");
+    //最初のインデックスをアクティブにする
+    i === 0 && dot.classList.add("is-active");
+    dot.addEventListener(
+      "click",
+      (e) => {
+        switchDot(e);
+        switchImgForDot(i);
+        switchDisableForBtn();
+        getCurrentPageNum();
+      },
+      false
+    );
```

Comment:
IMO
I felt I could also cut out the event function here.
```suggestion
    dot.addEventListener("click", (e) => clickDotEvents(e, i), false);
```
Cut out function

```js
const clickDotEvents = (e, i) => {
  switchDot(e);
  switchImgForDot(i);
  switchDisableForBtn();
  getCurrentPageNum();
};

```

Reply:
Thank you for reviewing.
It looks more easier and clearly, so I changed it :)
[204ea13](https://github.com/yuka830/js-lessons/pull/23/commits/204ea1369e0d7ffe21178b7689b29b2f89438245)